yarn/assets/configuration/spec.yaml

```diff
@@ -68,6 +86,24 @@ files:
             value:
               example: true
               type: boolean
+          - name: collect_apps_all_states
+            description: |
+              Enable this parameter when using `collect_app_metrics` to collect metrics from applications of all states.
+              For example, collect metrics from applications with 'NEW', 'SUBMITTED', 'FINISHED', and other states.
+            value:
+              example: false
+              type: boolean
+          - name: collect_apps_states_list
+            description: |
+              Configure the application states of which to collect application metrics.
+            value:
+              type: array
+              items:
+                type: string
+              example:
+                - "RUNNING"
+                - "FINISHED"
+                - "FAILED"
```

Comment:
You could even make it a list of enums, this way you won't have to validate them yourself. Example here: https://github.com/DataDog/integrations-core/blob/master/avi_vantage/assets/configuration/spec.yaml#L17-L33